spec/slack/real_time/client_spec.rb

```diff
@@ -159,16 +159,31 @@
       end
 
       describe '#run_handlers' do
-        context 'when store has no event hooks' do
+        before do
+          @events = client.store.class.events.dup
+          client.store.class.events.clear
+        end
+
+        after do
+          client.store.class.events.merge!(@events)
+        end
+
+        context 'when config#async_handlers is :all', if: ENV['CONCURRENCY'] == 'async-websocket' do
           before do
-            @events = client.store.class.events.dup
-            client.store.class.events.clear
+            client.async_handlers = :all
+            allow(socket).to receive(:run_async).and_return(Async.run { nil })
           end
 
           after do
-            client.store.class.events.merge!(@events)
+            client.async_handlers = :none
           end
 
+          it 'returns an Async::Task' do
+            expect(client.send(:run_handlers, 'example', {})).to be_a Async::Task
+          end
+        end
+
+        context 'when store has no event hooks' do
```

Comment:
you get an implicit test of the :none behavior here, since this test returns an empty array only when async_handlers is :none